kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/KroxyliciousClients.java

```diff
@@ -30,13 +30,15 @@
 
 class KroxyliciousClients {
     private final String bootstrapServers;
+    private final Map<String, Object> defaultClientConfiguration;
 
     private final List<Admin> admins;
-    private final List<Producer> producers;
-    private final List<Consumer> consumers;
+    private final List<Producer<?, ?>> producers;
+    private final List<Consumer<?, ?>> consumers;
 
-    KroxyliciousClients(String bootstrapServers) {
+    KroxyliciousClients(String bootstrapServers, Map<String, Object> defaultClientConfiguration) {
```

Comment:
I have a vague hope that this will mean we can drop the various with additional config methods from the tester and clients. 

Users who need to customise the client itself should probably be building and managing their own client

Reply:
I'm not sure about this.  I think we'd want use-cases sure as transactions or changing the ack mode supported.

Reply:
Yeah I can see that argument. 

My reasoning was something along the lines of tests that care about specific properties of the client should probably be explicit about them and not be using some magically configured client from the framework. 

Equally if we get the the API surface down to `producer(clusterName)` & `producer(clusterName, additionalConfig)` then we are probably doing ok.